File: PalindromeNumber/palindromeNumber.java
/*
 * Determine wheter an integer is a palindrome.
 * Do this without extra space.
 *
 */

import java.lang.*;

class palindromeNumber {
  public static void main(String[] args) {
    int inputInt = Integer.parseInt(args[0]);
    System.out.printf("Solution: %b\n", isPalindrome(inputInt));
  }

  public static boolean isPalindrome(int x) {
    int reversedNum = 0;
    int y = Math.abs(x);
    while(y > 0) {
      reversedNum = (reversedNum*10) + (y%10);
      y /= 10;
    }
    if( Math.abs(x) == reversedNum ) {
      return true;
    } else {
      return false;
    }
  }
}

File: PalindromeNumber/palindromeNumber.js
/**
 * @param {number} x
 * @return {boolean}
 *
 * Determine whether an integer is a palindrome.
 * Do this without extra space.
 *
 */
var isPalindrome = function(x) {
  if( x.toString() === x.toString().split("").reverse().join("")) {
    return true;
  } else {
    return false;
  }
};

console.log("Solution: ", isPalindrome(process.argv[2]));
